test/framework/ip/incluster.go

```diff
@@ -318,14 +318,16 @@ func (h *inCluster) claimIPAddress(ctx context.Context) (_ string, _ *ipamv1.IPA
 			return false, nil
 		}
 		if ip.Spec.Address == "" {
-			retryError = errors.Wrap(err, "IPAddress.Spec.Address is not set")
+			retryError = errors.New("IPAddress.Spec.Address is not set")
 			return false, nil
 		}
 
 		retryError = nil
 		return true, nil
 	})
 	if retryError != nil {
+		// Try best effort deletion of the unused claim before returning an error.
+		_ = h.client.Delete(ctx, claim)
```

Comment:
Ideally we would aggregate errors if .Delete also returns one instead of hiding this one

(maybe something for a follow-up)